pems/districts/views.py

```diff
@@ -1 +1,15 @@
-# Create your views here.
+from django.views.generic import TemplateView
+
+
+class IndexView(TemplateView):
+    template_name = "districts/index.html"
+
+
+class DistrictView(TemplateView):
+    template_name = "districts/district.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        district = self.kwargs.get("district")
+        context["district"] = district
+        return context
```

Comment:
So in the test below, the `district=1` kwarg is injected in `view.setup()`.

Here it is being pulled out of `kwargs` that are coming into `get_context_data()`. 

Is that how this works for a `TemplateView`? Do we get the `district=1` kwarg coming into `get_context_data()` "for free" when the appropriate route is called?

Reply:
Otherwise, I might expect the `district=1` kwarg to be injected in the call to `get_context_data()` in the test.

Reply:
Yep, I'm pretty sure that's how it works. According to the [documentation](https://docs.djangoproject.com/en/5.2/topics/class-based-views/generic-display/#dynamic-filtering) (not sure why it's not more visible, it's under _Dynamic Filtering_, but it is pretty useful and applies to other situations)

> when class-based views are called, various useful things are stored on self; as well as the request (self.request) this includes the positional (self.args) and name-based (self.kwargs) arguments captured according to the URLconf.

so we can get the `district=1` URL argument from `self.kwargs` "for free" in any method of the class-based view. To verify this I also stepped through the debugger and inspected `self.kwargs` and saw that the URL arguments are stored there.